client_option.go

```diff
@@ -1,22 +1,10 @@
 package getter
 
-import "context"
-
 // A ClientOption allows to configure a client
 type ClientOption func(*Client) error
 
 // Configure configures a client with options.
-func (c *Client) Configure(opts ...ClientOption) error {
-	if c.Ctx == nil {
-		c.Ctx = context.Background()
-	}
-	c.Options = opts
-	for _, opt := range opts {
-		err := opt(c)
-		if err != nil {
-			return err
-		}
-	}
+func (c *Client) Configure() error {
```

Comment:
Should this remain a public method?  How do consumers expect to use it?

Reply:
You are right; making it private for now. It's an artifact of the past 🙂.